src/Symfony/Component/Yaml/Parser.php

```diff
@@ -654,6 +654,9 @@ private function getNextEmbedBlock(int $indentation = null, bool $inSequence = f
             if ($this->isCurrentLineBlank()) {
                 $data[] = substr($this->currentLine, $newIndent);
                 continue;
+            } elseif ($this->isCurrentLineQuoteEnd()) {
```

Comment:
Won't this break cases where the current line does contain other characters before the terminating quotation character?

Reply:
And do we also need to take into account if the character matches the starting quotation character (and we are are actually parsing a quoted string at all here)?

Reply:
Thanks for your comments.

The method `isCurrentLineQuoteEnd()` is only checking if there is only one quote (' or ") in the trimmed line.

I tried to show with the invalid testcase, that the parser will fail when there is sth wrong with the quotes.

Yes, it's just a simple fix for the problems described in the issue. Maybe too simple? If there are lines with text in the quoted string, this lines have to be correctly indented. There is a bit the Problem, how the yaml parser is working. The qutotes with the text between are interpreted later in the process.

Reply:
Oh indeed, when writing the comment I mistakenly assumed that the new method was only checking if the current line ended with the quotation character.